remme/atomic_swap_tp/handler.py

```diff
@@ -0,0 +1,254 @@
+from remme.protos.token_pb2 import TransferPayload
+from remme.settings import SETTINGS_KEY_PUB_ENCRYPTION_KEY, SETTINGS_KEY_GENESIS_OWNERS, \
+    SETTINGS_SWAP_COMMISSION, ZERO_ADDRESS
+from remme.settings_tp.handler import _make_settings_key, _get_setting_value
+from remme.shared.basic_handler import BasicHandler, get_data
+from remme.shared.utils import hash256
+from remme.token_tp.client import TokenClient
+from remme.token_tp.handler import TokenHandler, get_account_by_address
+from remme.protos.certificate_pb2 import CertificateStorage, \
+    NewCertificatePayload, RevokeCertificatePayload, CertificateMethod
+from remme.shared.singleton import singleton
+
+LOGGER = logging.getLogger(__name__)
+
+FAMILY_NAME = 'AtomicSwap'
+FAMILY_VERSIONS = ['0.1']
+
+
+RANGE_ACCEPTANCE = 1
+
+# hours
+INTIATOR_TIME_LOCK = 24
+NON_INTIATOR_TIME_LOCK = 48
+
+INITIATOR_TIME_DELTA_LOCK = datetime.timedelta(hours=INTIATOR_TIME_LOCK)
+NON_INITIATOR_TIME_DELTA_LOCK = datetime.timedelta(hours=NON_INTIATOR_TIME_LOCK)
+
+
+@singleton
+class AtomicSwapHandler(BasicHandler):
+    def __init__(self):
+        super().__init__(FAMILY_NAME, FAMILY_VERSIONS)
+
+    def get_state_processor(self):
+        return {
+            AtomicSwapMethod.INIT: {
+                'pb_class': AtomicSwapInitPayload,
+                'processor': self._swap_init
+            },
+            AtomicSwapMethod.APPROVE: {
+                'pb_class': AtomicSwapApprovePayload,
+                'processor': self._swap_approve
+            },
+            AtomicSwapMethod.EXPIRE: {
+                'pb_class': AtomicSwapExpirePayload,
+                'processor': self._swap_expire
+            },
+            AtomicSwapMethod.SET_SECRET_LOCK: {
+                'pb_class': AtomicSwapSetSecretLockPayload,
+                'processor': self._swap_set_lock
+            },
+            AtomicSwapMethod.CLOSE: {
+                'pb_class': AtomicSwapClosePayload,
+                'processor': self._swap_close
+            },
+        }
+
+    def get_swap_info_from_swap_id(self, context, swap_id, to_raise_exception=True):
+        swap_info = get_data(context, AtomicSwapInfo, self.make_address_from_data(swap_id))
+        if to_raise_exception and not swap_info:
+            raise InvalidTransaction('Atomic swap was not initiated for {} swap id!'.format(swap_id))
+        return swap_info
+
+    def get_state_update(self, swap_info):
+        return {self.make_address_from_data(swap_info.swap_id): swap_info}
+
+    def get_datetime_from_timestamp(self, timestamp):
+        return datetime.datetime.fromtimestamp(timestamp)
+
+    def _swap_init(self, context, signer_pubkey, swap_init_payload):
+        """
+        if SecretLockOptionalBob is provided, Bob uses _swap_init to respond to requested swap
+        Otherwise, Alice uses _swap_init to request a swap and thus, Bob can't receive funds until Alice "approves".
+        TODO: check if receiver is a token address.
```

Comment:
This seems to be pretty important for usability, 
will you create a task in JIRA for this so we will implement it according to our process ?

Reply:
Implemented